### PR TITLE
Update: Option to allow afterthoughts in no-plusplus (fixes #4093)

### LINKED ISSUE #4093: Add exclusion for `for (var i = 0, i < l, i++)` in no-plus-plus
The no-plusplus rule is great, but I'd like to be able to allow people to still use a ++ in a for loop.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27276144-add-exclusion-for-for-var-i-0-i-l-i-in-no-plus-plus?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mrmckeb**

### CODE REVIEW

**Comment by platinumazure** [comment]
Afterthoughts? First I've heard of that term in this context.

**Comment by mrmckeb** [comment]
Hi @platinumazure, I had to research it a little, see an example of that usage here: https://en.wikipedia.org/wiki/For_loop. Mozilla refers to the `afterthought` as the `final-expression`.

File: docs/rules/no-plusplus.md [hunk 1]
```diff
@@ -32,16 +32,26 @@ j
 
 This rule is aimed at flagging the use of `++` and `--`. Some believe that the use of these unary operators reduces code quality and clarity. There are some programming languages that completely exclude these operators.
 
+### Options
+
+This rule, in it's default state, does not require any arguments. If you would like to enable one or more of the following you may pass an object with the options set as follows:
+
+* `forLoopAfterthoughts` set to `true` will allow you to use the unary operators `++` and `--` in the `afterthought` (or `final-expression`) of a `for loop`.
```
Comment (nzakas) [inline docs/rules/no-plusplus.md]: A little too much code text here. :)

`forLoopAfterthoughts` set to `true` will allow the use of the unary operators `++` and `--` in the afterthought (final expression) of a `for` loop.

Comment (nzakas) [inline docs/rules/no-plusplus.md]: I think the option probably needs to be named `allowForLoopAfterthoughts` to make it clear what it's doing.

Comment (mrmckeb) [inline docs/rules/no-plusplus.md]: Thanks @nzakas. I'll make those changes shortly.

**Comment by mrmckeb** [comment]
Hi @nzakas, this is now updated and the build is passing. Let me know if there are any other changes you'd like to see. Thanks again!

**Comment by nzakas** [comment]
LGTM. Thanks!

**Comment by mrmckeb** [comment]
Thank you!

**Comment by ericyang89** [comment]
thanks,  i meet the same problem

**Pull request merged**